### PR TITLE
events: use its own pgClient. simplified.

### PR DESCRIPTION
**Pull request by davidsparkles**
- [X] Give `events` its own PgClient
- [X] Simplify and typify `events`
- [X] Let `DbAppClient` and `DbGeneralPool` emit there own events
- [X] Let `DbGeneralPool` manage the pool scaling
- [X] Remove `db-auto-scaling`
- [X] Adjust `graceful-shutdown` to end `EventEmitter` after it sends the "exited" event

### CODE REVIEW

**Comment by dustin-H** [comment]
@davidsparkles Have you tested anything?

For example we could do the following tests:
- Autoscaling: Start two instances with a certain connection limit and see if both adjust their pool-size as expected.
- Autoscaling: Kill one instance and wait for the other instance to adjust the pool-size.
- Use events in the example project and test if they get emitted on multiple instances and the own one.

File: packages/server/lib/index.ts [hunk 1]
```diff
@@ -41,16 +40,19 @@ export class Server {
...
     this.bootKoa();
+    this.server = http.createServer(this.app.callback());
     bootLoader.addBootFunction(this.constructor.name, this.boot.bind(this));
+    gracefulShutdown.addServer(this.constructor.name, this.server);
   }
 
   private async boot(): Promise<void> {
     try {
       // start KOA on PORT
-      this.server = http.createServer(this.app.callback()).listen(this.serverConfig.port);
```
Comment (dustin-H) [inline packages/server/lib/index.ts]: Why have you moved this from the boot-function to constructor?

Is it necessary to run the terminus thing even before booting is finished?

**Comment by davidsparkles** [comment]
> @davidsparkles Have you tested anything?
> 
> For example we could do the following tests:
> 
>     * Autoscaling: Start two instances with a certain connection limit and see if both adjust their pool-size as expected.
> 
>     * Autoscaling: Kill one instance and wait for the other instance to adjust the pool-size.
> 
>     * Use events in the example project and test if they get emitted on multiple instances and the own one.

I tested all those things manually and they worked. There are no automated tests yet. I could create tests but this would include mocking the postgres db for `events` as well as `DbAppClient` and `DbGeneralPool`. So it would take some effort.

Are the successful manual tests sufficient for now, from you perspective @dustin-H ?

File: packages/server/lib/index.ts (hunk 1 above)
Comment (davidsparkles) [reply packages/server/lib/index.ts]: From my perspective, the constructor constructs everything that is needed and the boot function asynchronously boots and starts the things that have been constructed / created before. Thus, creating a server, configuring it and adding routes like `/healthcheck` is part of the constructing phase, whilst starting the server using `server.listen` belongs to the booting phase.

Do you disagree, @dustin-H ?

Comment (dustin-H) [reply packages/server/lib/index.ts]: I just wanted to understand.

I had a look at the Node.js specs. It is really just a constructor for that http.Server object, so its fine to do that inside the other constructor. :+1:

**Comment by dustin-H** [comment]
Yes, sure :)

I'm happy with manual tests, when it works.

**Pull request merged**